ccm_web/client/src/components/FileUpload.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { ChangeEvent, useRef } from 'react'
+import { Button, Card, CardContent, makeStyles, Typography } from '@material-ui/core'
+import CloudUploadIcon from '@material-ui/icons/CloudUpload'
+import { useSnackbar } from 'notistack'
+
+interface FileUploadProps {
+  onUploadComplete: (file: File) => void
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    textAlign: 'center'
+  },
+  cardContent: {
+    cursor: 'pointer'
+  },
+  actionsBar: {
+    backgroundColor: '#EEEEEE'
+  },
+  actionsBarIcon: {
+    verticalAlign: 'middle'
+  },
+  input: {
+
+  },
+  uploadIcon: {
+    color: '#3F648E'
+  }
+}))
+
+function FileUpload (props: FileUploadProps): JSX.Element {
+  const { enqueueSnackbar } = useSnackbar()
+  const handleFileSelected = (file: File|undefined): void => {
+    if (file === undefined) return
+    if (file.name.toUpperCase().match(/.+.CSV/) == null) {
+      enqueueSnackbar('File type not supported.', {
+        variant: 'error'
+      })
+      return
+    }
+    props.onUploadComplete(file)
+  }
+
+  const dragover = (e: React.DragEvent): void => {
+    e.preventDefault()
+  }
+  const dragdrop = (e: React.DragEvent): void => {
+    e.preventDefault()
+    handleFileSelected(e.dataTransfer.files[0])
+  }
+  const onFileInputClick = (e: React.MouseEvent<HTMLDivElement, MouseEvent>): void => {
+    fileInput.current?.click()
+    console.log(fileInput)
+    e.stopPropagation()
+  }
+
+  const fileInput = useRef<HTMLInputElement | null>(null)
+  const classes = useStyles()
+
+  const handleFileChange = (event: ChangeEvent<HTMLInputElement>): void => {
+    if (event.target === null || event.target.files === null) return
+    handleFileSelected(event.target.files[0])
+  }
+
+  return (
+    <div className={classes.root}>
+      <Card variant='outlined'>
+        <CardContent className={classes.cardContent} onDragOver={dragover} onDrop={dragdrop} onClick={(e) => onFileInputClick(e)}>
+          <div>
+            <CloudUploadIcon className={classes.uploadIcon} fontSize='large'/>
+            <Typography>
+              Drag and drop or <Button color='primary' component="span" onClick={(e: React.MouseEvent<HTMLDivElement, MouseEvent>) => onFileInputClick(e)}>browse your files</Button>
+            </Typography>
```

Comment:
I felt the Upload file area can be little bigger, since it help the user when they are dragging the file instead of using the upload option.
![Screen Shot 2021-05-06 at 7 31 08 AM](https://user-images.githubusercontent.com/8579775/117291614-39269500-ae3d-11eb-903f-084056c5148e.png)

Reply:
I agree. This was the approximate size during the usability test (helper text has been updated from the usability test
<img width="874" alt="Screen Shot 2021-05-06 at 8 48 50 AM" src="https://user-images.githubusercontent.com/46663379/117301673-dc30dc00-ae48-11eb-8acb-97c0a3258155.png">
):